markdown/renderer.go

```diff
@@ -23,26 +23,54 @@ var (
 	codeBlockChars          = []byte("```")
 	tableHeaderColChar      = []byte{'-'}
 	tableHeaderAlignColChar = []byte{':'}
+	heading1UnderlineChar   = []byte{'='}
+	heading2UnderlineChar   = []byte{'-'}
 )
 
 // Ensure compatibility with Goldmark parser.
 var _ renderer.Renderer = &Renderer{}
 
 // Renderer allows to render markdown AST into markdown bytes in consistent format.
 // Render is reusable across Renders, it holds configuration only.
-type Renderer struct{}
+type Renderer struct {
+	Opts *renderer.Config
```

Comment:
Hm, this would work but why not `underlineHeadersEnabled bool`? Would be more type - we can escape from this Goldmark string based weirdness as soon as we can ;p  

(We can eval this in `AddOptions` function

Reply:
yeah, but `renderer.Option` has `SetConfig` only called with `*renderer.Config`.

Best way, IMO - we can ignore Goldmark weirdness and just implement our own options entirely (`AddMarkdownftmOptions` whatever), and make goldmark's `AddOptions` on renderer empty. 

(which is basically what we did :) the code we have in master does nothing with the options)

And then we don't call `goldmark.WithRendererOptions`, becuase all that does is just apply the options, we can do it ourselves.